bevy_lint/tests/ui/zst_query/query.rs

```diff
@@ -0,0 +1,79 @@
+//! This tests the `zst_query` lint, specifically when triggered on the `Query` type.
+
+#![feature(register_tool)]
+#![register_tool(bevy)]
+#![deny(bevy::zst_query)]
+
+use bevy::prelude::*;
+use std::marker::PhantomData;
+
+#[derive(Component)]
+struct Foo;
+
+#[derive(Component)]
+#[allow(dead_code)]
+struct Bar(u32);
+
+#[derive(Component)]
+#[allow(dead_code)]
+struct Baz<T: Sized + Send + Sync + 'static>(T);
+
+#[derive(Component)]
+#[allow(dead_code)]
+struct Phantom<T>(PhantomData<T>);
```

Comment:
(Nit) Could you rename these types to be clearer what they do?

|Before|After|
|-|-|
|`Foo`|`ZST`|
|`Bar`|`SizedType` (there may be a better name for this)|
|`Baz`|`Generic`|
|`Phantom`|`Phantom`| (this is good already!)|